tests/binaries/heap-fastbins.c

```diff
@@ -2,29 +2,32 @@
  * -*- mode: c -*-
  * -*- coding: utf-8 -*-
  *
- * heap.c
+ * heap-fastbins.c
  *
  * @author: @_hugsy_
+ * @author: @_theguy147_
  * @licence: WTFPL v.2
  */
 
 #include <stdio.h>
-#include <stdint.h>
 #include <stdlib.h>
-#include <unistd.h>
-#include <string.h>
 
-int main(int argc, char** argv, char** envp)
-{
-        void* p1 = malloc(0x10);
-        void* p2 = malloc(0x20);
-        void* p3 = malloc(0x30);
-        memset(p1, 'A', 0x10);
-        memset(p2, 'B', 0x20);
-        memset(p3, 'C', 0x30);
-        free(p2);
-        __builtin_trap();
-        (void)p1;
-        (void)p3;
-        return EXIT_SUCCESS;
+void break_here() {
+	// GDB: `br *break_here`
+}
+
+int main(int argc, char** argv, char** envp) {
+	// allocate some chunks that have a suitable size for a fastbin
+	void* ptrs[10];
+	for (int i = 0; i < 10; i++) {
+		ptrs[i] = malloc(0x10);
+	}
+	// free 7 chunks to fill the tcache
+	for (int i = 0; i < 7; i++) {
+		free(ptrs[i]);
+	}
```

Comment:
@theguy147 In `test_cmd_heap_bins_non_main`, we use `set environment GLIBC_TUNABLES glibc.malloc.tcache_count=0` to eliminate the tcache. It looks this change has the same effect, but it might be more consistent to use the same method when we want to ignore the tcache in a test.

Reply:
oh, ok, i didnt notice this. That makes sense as well. But then we should at least add a comment to the c file explaining this.

Reply:
ok, i rolled this change back now